### ISSUE TITLE
"Show Irrelevant Dates" results in removal of due date

### ISSUE THREAD

**Issue by Etherealflux** [Submitty/Submitty#4480]
**Describe the bug**
Pressing the "Show Irrelevant Dates" button on the *Dates* tab of a gradeable configuration causes the "Will this assignment have a due date?" toggle to default to "no", even if the assignment has a due date configured. This saves immediately, causing the assignment to lose its due date.

**Expected behavior**
The radio buttons should reflect the current setting.

**To Reproduce**
1. Go to a gradeable's configuration page (I was editing an existing one) 
2. Go to the Dates tab
3. Click "Show Irrelevant Dates"
4. The due date is disabled

**Configuration**
- Chrome 63

**Screenshots**
N/A

**Additional context**
Grade inquiries are disabled. The assignment is not yet open to students - it is in the "BETA - opening for testing by TAs" state.

**Comment by MasterOdin** [comment]
What are the selected settings for "What are students allowed to do?" on the Submissions/Autograding tab?

**Comment by Etherealflux** [comment]
Ahaha, I just realized how far out of date that browser is. Must not be auto-updating on Linux.

It does behave differently in Chrome 76. If I click "Show All Dates", the toggle is set incorrectly, but it does not save. However, if I set it to "Yes" and then hit "Hide Irrelevant Dates", it *still* removes the due date.

I confirmed that this happened in Incognito mode, so it shouldn't be a rogue extension.

It looks like the problem was that we, mistakenly, had it set to *not* allow students to view the gradeable. This, naturally, made a due date irrelevant, so it was hidden by default. Only in this case does the erroneous behavior occur.

**Comment by MasterOdin** [comment]
Yeah, I think it's more accurate to say that the bug here would be that this button is not disabled as well as being auto off when student submission is disabled. Thanks for your help in determining what's going on.
